### PR TITLE
RT-23 init server core + RT-24 client connection

### PR DESCRIPTION
**Pull request by Valegox**
- Setup `ServerNetworkHandler` and `ClientNetworkHandler`. They both encapsulate `boost::asio` and receive requests asynchronously. The async process requires to create a thread inside each handler, but callbacks stay on the main thread. The `ServerNetworkHandler` stores its client endpoints refered by their entity id.

- Setup a `RTypeProtocol` lib common to RType server and client. It defines `ServerToClientPacket` / `ClientToServerPacket` structs and the `GameEvent` class.

Server launch and client connection can be tested by running each program. For testing, the client sends 3 messages to the server (see `src/client/main.c`).

Serialization and call to ECS systems have been implemented as well.

### CODE REVIEW

**Review by Beafowl-Pull — DISMISSED** [review]

**Review by Beafowl-Pull — DISMISSED** [review]
RAS

**Review by Beafowl-Pull — DISMISSED** [review]
RAS x2

File: src/client/CMakeLists.txt [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
         ${Boost_LIBRARIES}
+        Boost::serialization
+        Threads::Threads
+        ecs
+        rtype_protocol
+        Boost::asio
+        Boost::serialization
 )
 
+target_include_directories(client PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/network")
```
Comment (GlassAlo) [inline src/client/CMakeLists.txt]: Might be better to put the include in the respective folders using the CMakeLists' folders

File: src/client/CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+target_include_directories(client PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../rtype_protocol")
```
Comment (GlassAlo) [inline src/client/CMakeLists.txt]: same as for line 24

File: src/ecs/CMakeLists.txt [hunk 3]
```diff
@@ -17,6 +17,7 @@ target_link_libraries(ecs
 )
 
 target_include_directories(ecs PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(ecs PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/Event")
```
Comment (GlassAlo) [inline src/ecs/CMakeLists.txt:20]: Is this useful ? it should be in the CMakeList from the Event Folder

**Review by GlassAlo — DISMISSED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -9,6 +9,10 @@ if (NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 20)
 endif()
 
+if (GCC)
```
Comment (Beafowl-Pull) [inline CMakeLists.txt]: remove this, it's useless

**Review by Beafowl-Pull — CHANGES_REQUESTED** [review]
Just a thing but it's globally good

**Review by Beafowl-Pull — APPROVED** [review]
RAS

**Pull request merged**